### ISSUE TITLE
S3 presigned url signature changed if the key contains consecutive '/'

### ISSUE THREAD

**Issue by zhouzyue** [aws/aws-sdk-java-v2#4819]
### Describe the bug

After upgrading the sdk version from `2.20.103` to `2.22.9`, we found that the signature had changed and encountered `SignatureDoesNotMatch` errors when the requested object keys contain multiple consecutive '/'.

### Expected Behavior

The generated presigned url should still work

### Current Behavior

Encountered `SignatureDoesNotMatch` error when using the presigned url generated by new version.

### Reproduction Steps

The following test can pass with `2.20.103` but not with `2.22.9`.

```
import org.junit.jupiter.api.Test;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.s3.presigner.S3Presigner;
import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
import java.time.Clock;
import java.time.Duration;
import java.time.Instant;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static org.mockito.Mockito.CALLS_REAL_METHODS;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.mockStatic;
import static org.mockito.Mockito.when;

class S3PresignerTest {

    @Test
    void test() {
        var request = GetObjectPresignRequest.builder()
            .signatureDuration(Duration.ofHours(1))
            .getObjectRequest(b -> b.bucket("bucket").key("a//test.jpg"))
            .build();

        var fakeNow = Instant.ofEpochMilli(0);

        var fakeClock = mock(Clock.class);
        when(fakeClock.instant()).thenReturn(fakeNow);

        var awsCredentialsProvider =
            StaticCredentialsProvider.create(AwsBasicCredentials.create("accessKey", "secretKey"));

        try (
            var instant = mockStatic(Instant.class, CALLS_REAL_METHODS);
            var clock = mockStatic(Clock.class, CALLS_REAL_METHODS)
        ) {
            instant.when(Instant::now).thenReturn(fakeNow);
            clock.when(Clock::systemUTC).thenReturn(fakeClock);

            var presigner = S3Presigner.builder()
                .credentialsProvider(awsCredentialsProvider)
                .region(Region.EU_CENTRAL_1)
                .build();
            var presignedUrl = presigner.presignGetObject(request).url();

            assertThat(presignedUrl)
                .hasProtocol("https")
                .hasHost("bucket.s3.eu-central-1.amazonaws.com")
                .hasPath("/a//test.jpg")
                .hasParameter("X-Amz-Algorithm", "AWS4-HMAC-SHA256")
                .hasParameter("X-Amz-Date", "19700101T000000Z")
                .hasParameter("X-Amz-SignedHeaders", "host")
                .hasParameter("X-Amz-Expires", "3600")
                .hasParameter("X-Amz-Credential", "accessKey/19700101/eu-central-1/s3/aws4_request")
                .hasParameter("X-Amz-Signature", "7c7c5a9b0e4a002b5b7e7c067f91f1148418fcb2b2c3e3112a409e4ac7bf515a");
        }
    }
}
```

### AWS Java SDK version used

2.22.9

### JDK version used

17

### Operating System and version

MacOs 14.1.2

**Comment by debora-ito** [comment]
@zhouzyue thank you for reaching out, I can reproduce the problem. The error started with version `2.21.16`. We are looking into it.

On a side note, I don't recommend to have a unit test to check the url parameters like the one you provided, the test can start to fail if S3 decides to change their headers in the future.

**Comment by debora-ito** [comment]
@zhouzyue a fix was released as part of version `2.23.0`, please try it out and let us know if you have any other question.
